<launch>


    <!-- MASTER NODE! -->
    <node name="cpi_compare" pkg="cpi_compare" type="estimator" output="screen" clear_params="true" required="true">

        <!-- simulation parameters -->
        <param name="simPath"      type="string" value="$(find cpi_compare)/../cpi_simulation/GAZEBO_FREQ_100/rawdata_01/" />
        <param name="useSerialSIM" type="bool"   value="true" />
        <rosparam param="gravity">[0.0,0.0,9.8]</rosparam>

        <!-- Noise Values for ADIS16448 Sensor -->
        <param name="sigma_camera"                 type="double"   value="0.00206555403" />
        <param name="gyroscope_noise_density"      type="double"   value="0.005" />
        <param name="gyroscope_random_walk"        type="double"   value="4.0e-06" />
        <param name="accelerometer_noise_density"  type="double"   value="0.01" />
        <param name="accelerometer_random_walk"    type="double"   value="0.0002" />

        <!-- Noise Values for ETH Dataset Sensor -->
        <!--<param name="sigma_camera"                 type="double"   value="0.00206555403" />-->
        <!--<param name="gyroscope_noise_density"      type="double"   value="1.6968e-03" />-->
        <!--<param name="gyroscope_random_walk"        type="double"   value="1.9393e-05" />-->
        <!--<param name="accelerometer_noise_density"  type="double"   value="2.0000e-3" />-->
        <!--<param name="accelerometer_random_walk"    type="double"   value="3.0000e-3" />-->

        <!-- sensor rates -->
        <param name="imurate"   type="int"  value="100" />
        <param name="camrate"   type="int"  value="10" />

        <!-- SYNTH DATASET IMU WAITING NUM -->
        <param name="imuwait"   type="int"  value="300" /> <!-- 100hz gazebo freq -->
        <!--<param name="imuwait"   type="int"  value="900" /> &lt;!&ndash; 200hz gazebo freq &ndash;&gt;-->
        <!--<param name="imuwait"   type="int"  value="2300" /> &lt;!&ndash; 400hz gazebo freq &ndash;&gt;-->
        <!--<param name="imuwait"   type="int"  value="4900" /> &lt;!&ndash; 800hz gazebo freq &ndash;&gt;-->

        <!-- config flags -->
        <param name="useGroundTruthInit"    type="bool"   value="true" />
        <param name="useInverseDepth"       type="bool"   value="false" />

        <!-- numerical parameters -->
        <param name="minPoseFeatureInit"    type="int"   value="8" />
        <param name="lagSmootherAmount"     type="int"   value="3" />
        <param name="uvWindowSize"          type="int"   value="22" />


    </node>


</launch>
